crates/oxc_linter/src/rules/vitest/require_local_test_context_for_concurrent_snapshots.rs

```diff
@@ -0,0 +1,257 @@
+use oxc_ast::{
+    ast::{BindingPatternKind, FormalParameters, PropertyKey},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_semantic::AstNode;
+use oxc_span::Span;
+
+use crate::{
+    context::LintContext,
+    rule::Rule,
+    utils::{
+        is_type_of_jest_fn_call, parse_expect_jest_fn_call, JestFnKind, JestGeneralFnKind,
+        PossibleJestNode,
+    },
+};
+
+fn require_local_test_context(span0: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("require local Test Context for concurrent snapshot tests".to_string())
+        .with_help("Use local Test Context instead")
+        .with_label(span0)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct RequireLocalTestContextForConcurrentSnapshots;
+
+declare_oxc_lint!(
+    /// ### Examples
+    ///
+    /// ```javascript
+    /// // invalid
+    /// test.concurrent('myLogic', () => {
+    ///     expect(true).toMatchSnapshot();
+    /// })
+    ///
+    /// describe.concurrent('something', () => {
+    ///     test('myLogic', () => {
+    ///         expect(true).toMatchInlineSnapshot();
+    ///     })
+    /// })
+    /// ```
+    ///
+    /// ```javascript
+    /// // valid
+    /// test.concurrent('myLogic', ({ expect }) => {
+    ///     expect(true).toMatchSnapshot();
+    /// })
+    ///
+    /// test.concurrent('myLogic', (context) => {
+    ///     context.expect(true).toMatchSnapshot();
+    /// })
+    /// ```
+    RequireLocalTestContextForConcurrentSnapshots,
+    style,
+);
+
+impl Rule for RequireLocalTestContextForConcurrentSnapshots {
+    fn run_once(&self, ctx: &LintContext) {
```

Comment:
Looks like we don't need to collect function parameters? We can iterate all `expect` call expression and combine you `check` method  

https://github.com/veritem/eslint-plugin-vitest/blob/8b11357b8690c276b869a04140f5238679bab5de/src/rules/require-local-test-context-for-concurrent-snapshots.ts#L33